src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -500,6 +597,144 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+                // put the receiver into a temp, and then store all the arguments up to the conversion into their own
+                // temps.
+
+                if (requiresInstanceReceiver)
+                {
+                    Debug.Assert(visitedReceiver is not null);
+                    visitedReceiver = _factory.StoreToTemp(visitedReceiver, out var receiverStore, refKind: visitedReceiver.GetRefKind());
+                    storesToTemps.Add(receiverStore);
+                }
+
+                int currentIndex = 0;
+
+                while (currentIndex < arguments.Length)
+                {
+                    Debug.Assert(arguments[interpolatedStringHandlerArgumentIndex] is BoundConversion { ConversionKind: ConversionKind.InterpolatedStringHandler });
+
+                    // Store up to the index of the conversion
+                    VisitAndStoreToTemps(
+                        expanded,
+                        argsToParamsOpt,
+                        parameters,
+                        argumentRefKindsOpt,
+                        arguments,
+                        forceLambdaSpilling: true,
+                        startIndex: currentIndex,
+                        length: interpolatedStringHandlerArgumentIndex - currentIndex,
+                        argumentsAreVisited,
+                        rewrittenArguments: actualArguments,
+                        refKinds: refKinds,
+                        storesToTemps: storesToTemps,
+                        tempsBuilder: temporariesBuilder);
+
+                    var interpolatedString = (BoundInterpolatedString)((BoundConversion)arguments[interpolatedStringHandlerArgumentIndex]).Operand;
+                    Debug.Assert(interpolatedString.InterpolationData.HasValue);
+                    var interpolationData = interpolatedString.InterpolationData.GetValueOrDefault();
+
+                    BoundLocal? trailingOutVariable = null;
+                    var constructor = interpolationData.GetValidConstructor();
+                    var lastArgument = constructor.Arguments[^1];
+                    if (interpolationData.HasTrailingHandlerValidityParameter)
+                    {
+                        var outValidityTemp = _factory.SynthesizedLocal(lastArgument.Type!, lastArgument.Syntax);
+                        temporariesBuilder.Add(outValidityTemp);
+                        trailingOutVariable = _factory.Local(outValidityTemp);
+                    }
+
+                    foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                    {
+                        // We can have duplicates because arguments can be duplicated, but we
+                        // only need to register one placeholder per argument
+                        if (HasPlaceholderReplacement(placeholder))
+                        {
+                            continue;
```

Comment:
>continue [](http://example.com/codeflow?start=28&length=8)

Is this code reachable? It looks like we are allocating distinct instances for each parameter, even if they are going to refer to the same argument. What am I missing?